pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -450,3 +451,40 @@ func (a Adapter) Delete(labels map[string]string) error {
 
 	return a.Client.DeleteDeployment(labels)
 }
+
+func (a Adapter) Log(follow, debug bool) (io.ReadCloser, error) {
+
+	if !utils.ComponentExists(a.Client, a.ComponentName) {
+		return nil, errors.Errorf("the component %s doesn't exist on the cluster", a.ComponentName)
+	}
```

Comment:
I think we don't need this check if we just check for the status of the pod https://github.com/openshift/odo/pull/3384/files#r451283469

Reply:
update to use `GetPodUsingComponentName`, now not checking the status.